### PR TITLE
Feature/jsonls avoid duplicate prints

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/62439

Modifies the JSON-LD abstract class to avoid returning data for already fetched posts.
This is to avoid re-printing JSON-LS data for the same post multiple times on the page (no-no in Google standards)

### CODE REVIEW

File: src/Tribe/JSON_LD/Abstract.php [hunk 1]
```diff
@@ -42,27 +49,47 @@ public static function instance( $name = null ) {
 
 	/**
 	 * Compile the schema.org event data into an array
+	 *
+	 * @param mixed $post            Either a post ID or a WP_post object.
+	 * @param array $args            {
+	 *                               Optional. An array of arguments to control the returned data.
```
Comment (barryhughes) [inline src/Tribe/JSON_LD/Abstract.php]: This looks just slightly weird. Can we format per [this example](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-1-parameters-that-are-arrays) (ie, indent the array args)?

File: src/Tribe/JSON_LD/Abstract.php [hunk 2]
```diff
@@ -126,18 +154,79 @@ public function markup( $post = null, $args = array() ) {
...
+
+		self::$fetched_post_ids[] = $post->ID;
+	}
+
+	public function reset_fetched_post_ids() {
+		self::class_reset_fetched_post_ids();
+	}
+
+	public function unset_fetched_post_id( $post ) {
+		$post = $this->get_post_object( $post );
```
Comment (barryhughes) [inline src/Tribe/JSON_LD/Abstract.php]: We're really only interested in getting the ID here: is it possible to drop the `get_post_object()` method and call `Tribe__Main::post_id_helper()` directly?

File: tests/wpunit/Tribe/JSON_LD/AbstractTest.php [hunk 3]
```diff
@@ -0,0 +1,176 @@
+<?php
+namespace Tribe\JSON_LD;
+
+require_once codecept_data_dir( 'classes/Tribe__JSON_LD__Test_Class.php' );
+
+use Tribe__JSON_LD__Test_Class as Jsonld;
+
+class AbstractTest extends \Codeception\TestCase\WPTestCase {
```
Comment (barryhughes) [inline tests/wpunit/Tribe/JSON_LD/AbstractTest.php]: Maybe switch to `Snake_Case` here?

Comment (lucatume) [inline tests/wpunit/Tribe/JSON_LD/AbstractTest.php]: We have this format throughout all of the test code, Codeception will generate class names in he `Whatever_Class_Name__IsTest` format. To have it like that we should update all of the `codeception.yml` and tests to stick.

Comment (barryhughes) [inline tests/wpunit/Tribe/JSON_LD/AbstractTest.php]: OK, just feels weird to not follow our own coding standards for tests. `Tribe__Template_Checker_Test` and `Logging_Test` aren't camelcased but seem to work ok?

**Comment by barryhughes** [comment]
:fireworks:

**Pull request merged**